components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/http/management/HTTPEndpointManager.java

```diff
@@ -63,19 +63,28 @@
             new ConcurrentHashMap<String, ConcurrentHashMap<Integer, Pattern>>();
 
     private int internalInboundPort;
-
-    private InternalAPIDispatcher internalAPIDispatcher;
-
-    private boolean internalApiEnabled;
+    private int internalInboundHttpsPort;
+    private InternalAPIDispatcher internalHttpAPIDispatcher;
+    private InternalAPIDispatcher internalHttpsAPIDispatcher;
+    private boolean internalHttpApiEnabled;
+    private boolean internalHttpsApiEnabled;
 
     private HTTPEndpointManager() {
         super();
-        internalInboundPort = ConfigurationLoader.getInternalInboundPort();
+
+        internalInboundPort = ConfigurationLoader.getInternalInboundHttpPort();
+        internalInboundHttpsPort = ConfigurationLoader.getInternalInboundHttpsPort();
         if (internalInboundPort != -1) {
-            internalInboundPort += PersistenceUtils.getPortOffset();
-            List<InternalAPI> internalAPIList = ConfigurationLoader.loadInternalAPIs(Constants.INTERNAL_APIS_FILE);
-            internalAPIDispatcher = new InternalAPIDispatcher(internalAPIList);
-            internalApiEnabled = !internalAPIList.isEmpty();
+            int portOffset = PersistenceUtils.getPortOffset();
+            internalInboundPort += portOffset;
```

Comment:
Shall we refactor as internalInboundPort = internalInboundPort + portOffset for better readability?